metadata_backend/api/handlers/submission.py

```diff
@@ -536,29 +541,50 @@ async def publish_submission(self, req: Request) -> Response:
             await self.metax_handler.check_connection()
 
         submission = await operator.read_submission(submission_id)
+        # Validate that submission seems to have required data for publishing
+        if "doiInfo" not in submission:
+            raise web.HTTPBadRequest(reason=f"Submission '{submission_id}' must have the required DOI metadata.")
+        has_study = False
+        # has_ega_dac = False
+        for _obj in submission["metadataObjects"]:
+            accession_id = _obj["accessionId"]
+            schema = _obj["schema"]
+            object_data, _ = await obj_op.read_metadata_object(schema, accession_id)
+            if schema == "study":
+                has_study = True
+            # if schema == "dac":
+            #     has_ega_dac = True
+            if self.rems_handler.enabled and "dac" not in submission:
+                raise web.HTTPBadRequest(reason=f"Submission '{accession_id}' must have DAC.")
+        if not has_study:
+            raise web.HTTPBadRequest(reason=f"Submission '{submission_id}' must have a study.")
+        # if not has_ega_dac:
+        #     raise web.HTTPBadRequest(reason=f"Submission '{submission_id}' must have an EGA DAC.")
```

Comment:
```suggestion
        for _obj in submission["metadataObjects"]:
            accession_id = _obj["accessionId"]
            schema = _obj["schema"]
            object_data, _ = await obj_op.read_metadata_object(schema, accession_id)
            if schema == "study":
                has_study = True
            if self.rems_handler.enabled and "dac" not in submission:
                raise web.HTTPBadRequest(reason=f"Submission '{accession_id}' must have DAC.")
        if not has_study:
            raise web.HTTPBadRequest(reason=f"Submission '{submission_id}' must have a study.")
```
remove commented functionality if it's no longer needed